sqlmesh/core/engine_adapter/base.py

```diff
@@ -67,6 +68,32 @@ def is_insert_overwrite(self) -> bool:
     def is_replace_where(self) -> bool:
         return self == InsertOverwriteStrategy.REPLACE_WHERE
 
+    @property
+    def requires_condition(self) -> bool:
+        return self.is_replace_where or self.is_delete_insert
+
+
+@dataclass
+class SourceQuery:
+    generator: t.Iterator[Query]
+    columns_to_types: t.Optional[t.Dict[str, exp.DataType]] = None
+    batched: bool = False
+    is_from_df: bool = False
+
+    @property
+    def single_query(self) -> t.Iterator[Query]:
+        assert not self.batched
+        yield from self.generator
+
```

Comment:
@eakmanrq is there an example in this PR where the generator yields more than one query?

Reply:
https://github.com/TobikoData/sqlmesh/blob/659a972b207abadc9796f44bdb45f64c1598839d/sqlmesh/core/engine_adapter/base.py#L175